### PR TITLE
PSP-7884 add take completion date.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

**Review by asanchezr — APPROVED** [review]

File: source/backend/api/Constants/TakeStatusTypes.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+
+namespace Pims.Api.Constants
+{
+    [JsonConverter(typeof(JsonStringEnumMemberConverter))]
+    public enum TakeStatusTypes
```
Comment (eddherrera) [inline source/backend/api/Constants/TakeStatusTypes.cs:7]: Please use AcquisitionTakeStatusTypes, namespace Pims.Api.Models.CodeTypes

File: source/frontend/src/features/mapSideBar/property/tabs/takes/update/TakeSubForm.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { SectionField } from '@/components/common/Section/SectionField';
 import AreaContainer from '@/components/measurements/AreaContainer';
 import { Roles } from '@/constants';
 import * as API from '@/constants/API';
+import { TakesStatusTypes } from '@/constants/takesStatusTypes';
```
Comment (eddherrera) [inline source/frontend/src/features/mapSideBar/property/tabs/takes/update/TakeSubForm.tsx:13]: Please use ApiGen_CodeTypes_AcquisitionTakeStatusTypes

File: source/frontend/src/features/mapSideBar/property/tabs/takes/update/TakeSubForm.tsx [hunk 3]
```diff
@@ -54,6 +55,12 @@ const TakeSubForm: React.FunctionComponent<ITakeSubFormProps> = ({
   );
   const takeStatusTypeCode = getIn(values, withNameSpace(nameSpace, 'takeStatusTypeCode'));
 
+  React.useEffect(() => {
+    if (currentTake.completionDt && currentTake.takeStatusTypeCode !== TakesStatusTypes.COMPLETE) {
```
Comment (eddherrera) [inline source/frontend/src/features/mapSideBar/property/tabs/takes/update/TakeSubForm.tsx:59]: same as previous comment, use Generated models "ApiGen_CodeTypes_AcquisitionTakeStatusTypes"

File: source/frontend/src/features/mapSideBar/property/tabs/takes/update/TakeSubForm.tsx [hunk 4]
```diff
@@ -121,6 +128,20 @@ const TakeSubForm: React.FunctionComponent<ITakeSubFormProps> = ({
           disabled={!canEditTake}
         />
       </SectionField>
+      <SectionField
+        label="Completion date"
+        required={currentTake.takeStatusTypeCode === TakesStatusTypes.COMPLETE}
+        tooltip={`This will be enabled when the file status is set to "Completed"`}
```
Comment (eddherrera) [inline source/frontend/src/features/mapSideBar/property/tabs/takes/update/TakeSubForm.tsx:134]: no need for brackets

**Review by eddherrera — CHANGES_REQUESTED** [review]
Small updates regarding the use of the code type models.

**Review by eddherrera — APPROVED** [review]
Please just replace models.

**Pull request merged**